evcouplings/utils/pipeline.py

```diff
@@ -472,6 +476,89 @@ def _handler(signal_, frame):
         raise
 
 
+def calculate_memory_requirements(config):
+    """
+
+    Calculates the memory requirements in MB based on the pipeline configuration
+
+    memory_in_MB = (1/2 * q^2 * (L - 1) * L  + q * L) / 12500
+
+    with:
+        - q: states of Potts model
+        - L: Sequence length
+
+
+    Parameters
+    ----------
+    config : dict
+        A dictionary holding the pipeline configurations
+
+    Returns
+    -------
+    int
+        RAM requirements in MB
+
+    Raises
+    ------
+    ResourceError
+    """
+
+    # gather alignment stages (monomer has only one, complex has 2)
+    stages = filter(lambda x: "align" in x, config["stages"])
+    L = 0
+
+    # extract state information
+    # if alphabet is not explicitly defined assume amino acid alphabet + gap
+    q = len(config["couplings"]["alphabet"]) if config["couplings"]["alphabet"] else 21
+    if config["couplings"]["ignore_gaps"]:
+        q -= 1
+
+    # iterate through alignment stages to determine input sequence length
+    global_cfg = config["global"]
+    for stage in stages:
+        incfg = {
+            **config[stage],
+            **global_cfg
+        }
+
+        # three cases have to be considered
+        # 1) The range is specified and we can immediately calculate the input length
+        # 2) An alignment file is specified; so we have to read in the first sequence to extract the length
+        # 3) An identifier is specified; so we have to download the sequence
+        if incfg["region"]:
+            L += incfg["region"][1] - incfg["region"][0]
+
+        elif incfg["sequence_file"]:
+            # first try to autodetect format of alignment
+            with open(incfg["sequence_file"]) as f:
+                format = detect_format(f)
+                if format is None:
+                    raise InvalidParameterError(
+                        "Format of input alignment {} could not be "
+                        "automatically detected.".format(
+                            incfg["sequence_file"]
+                        )
+                    )
+
+            # read alignment to get length
+            # alignment might still be in non-focus mode
+            # we just ignore that for now. We do not need an exact calculation of the RAM requirements, an upper limit
```

Comment:
The sequence length before focussing the sequence might be a loooot longer due to inserts so the memory requirement could be vastly overestimated if one doesn't focus.